src/App.tsx

```diff
@@ -1,14 +1,65 @@
-import React from 'react';
+import debounce from 'lodash.debounce';
+import { useMemo, useCallback, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { List } from './components/List/List';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setApplyQuerry] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState('');
+  const [isFocused, setIsFocused] = useState(false);
+
+  const reset = () => {
+    setSelectedPerson('');
+  };
+
+  const handleValue = () => {
+    let value = query;
+
+    if (selectedPerson) {
+      value = selectedPerson;
+    }
+
+    return value;
+  };
```

Comment:
what's the purpose of this function? you returning person or query but in general it's value?? looks redundant and should be refactored

Reply:
This function sets value of input(should it be query or selectedPerson)

Reply:
if selectedValue true value of input is SelectedValue else it is query

Reply:
consider using a ternary operator instead